### ISSUE TITLE
Granular blocking of WASM

### ISSUE THREAD

**Issue by emanruse** [uBlockOrigin/uMatrix-issues#226]
A recent [feedback on uBO issue](https://github.com/uBlockOrigin/uBlock-issues/issues/900#issuecomment-589777075) explained that currently no browser provides per site possibility to block WASM. 

Considering:

https://www.infoq.com/news/2019/10/WebAssembly-wasm-malicious-usage/
https://www.dimva2019.org/wp-content/uploads/sites/31/2019/06/DIMVA19-slides-2-R.pdf

it might be good uMatrix to be able to provide this possibility (if it is technically possible).

**Comment by gorhill** [comment]
uBO already falls back to JS when WebAssembly is not possible, I will consider this issue invalid.

**Comment by gorhill** [comment]
Sorry, given that you linked to an unrelated issue, I misunderstood what you asked here. If I understand correctly, you are suggesting ability to block WASM specifically, on a per-site basis.

If so, I would say that if you do not trust a site to execute WASM, than there is little reason to trust the same site to execute JS, which you can already block through rules.

**Comment by emanruse** [comment]
Yes, that is the suggestion and you are right: if one doesn't trust a website one has no reason to allow it to run any code (JS or WASM).

But considering that the study is about WASM explicitly and not about JS (which I guess has its own issues, especially those related to Spectre etc): is it wrong to assume that enabling JS but blocking WASM *may* provide some additional security? (I am not aware if WASM has more or less access to local resources like RAM, files etc)
